### ISSUE TITLE
Is it possible to add HLint rule to require INLINABLE

### ISSUE THREAD

**Issue by phadej** [ndmitchell/hlint#994]
Say I have type-class

```haskell
class Monad m => Foo m where
   foo :: m ...
```

I'd want to require that all `foo = ` bindings have `{-# INLINABLE foo #-}` (or `INLINE`) in the same scope. AFAIU this cannot be implement as user hint.

I'll be perfectly happy if all `foo =` should have `{-# INLINABLE foo #-}` attached, independencly if they are actually instance methods or not. It would be my problem to turn false positives (I doubt I'll have local variables named `foo` e.g.).

**Comment by ndmitchell** [comment]
With the user configurable .yaml file it seems entirely reasonable to require certain bindings to have certain pragmas. They are currently known as restrictions, and in a way they still are - just restricting the kind of pragmas you can omit.
